### PR TITLE
EZP-30775: PHP API: Managing

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZP-30775
| Versions      | n/a

Main changes:

- disconnects the page from the CookBook bundle and updates code examples
- adds examples covering Bookmarks, Languages, Workflow and Object states
- removes the "Converting request parameters into API objects" cookbook page (its content is already covered in practice in examples throughout)
- cleans up the text

**Note for reviewers**: `public_php_api_creating_content.md` was not touched in this PR, the file was only renamed (which confused GH), please don't review.

*Note: branch php-api serves as a feature branch, the whole rewrite of PHP API section will be merged to it, and then to 2.5 at once.*

### CODE REVIEW

File: docs/api/public_php_api_creating_content.md [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+``` php
+$locationCreateStruct = $locationService->newLocationCreateStruct( 2 );
+```
+
+### Creating and publishing
+
+To actually create your Content in the Repository, you need to use `ContentService::createContent()`. This method expects a `ContentCreateStruct`, as well as a `LocationCreateStruct`. You have created both in the previous steps.
+
+``` php
+$draft = $contentService->createContent( $contentCreateStruct, array( $locationCreateStruct ) );
```
Comment (mikadamczyk) [inline docs/api/public_php_api_creating_content.md]:
```suggestion
$draft = $contentService->createContent($contentCreateStruct, [$locationCreateStruct]);
```

File: docs/api/public_php_api_creating_content.md [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+$contentInfo = $contentService->loadContentInfo( $contentId );
+$contentDraft = $contentService->createContentDraft( $contentInfo );
+```
+
+To create your draft, you need to load the Content item's ContentInfo using `ContentService::loadContentInfo()`. You can then use `ContentService::createContentDraft()` to add a new Draft to your Content.
+
+``` php
+// instantiate a content update struct and set the new fields
+$contentUpdateStruct = $contentService->newContentUpdateStruct();
+$contentUpdateStruct->initialLanguageCode = 'eng-GB'; // set language for new version
```
Comment (mikadamczyk) [inline docs/api/public_php_api_creating_content.md]: This comment could be a part of article

File: docs/api/public_php_api_creating_content.md [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+The PHP API consumer is responsible for creating a Content item version draft and publishing it after translation removal.
+
+Since the returned Content draft is to be published, both search and HTTP cache are already handled
+by `PublishVersion` slots once the call to `publishVersion()` is made.
+
+Example:
+
+``` php
+$repository->beginTransaction();
+/** @var \eZ\Publish\API\Repository\Repository $repository */
```
Comment (mikadamczyk) [inline docs/api/public_php_api_creating_content.md]: This line should be the first one

File: docs/api/public_php_api_creating_content.md [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+
+As explained above, the `setField()` method can accept various values: an instance of the Field Type's Value class, a primitive type, or a hash. The last two depend on what the `Type::acceptValue()` method is build up to handle. TextLine can, for instance, accept a simple string as an input value. In this example, you will see how to set an Image value.
+
+Let's assume that you use the default image class. Creating your Content, using the Content Type and a `ContentCreateStruct`, has been covered above, and can be found in the full code. Let's focus on how the image is provided.
+
+``` php
+$file = '/path/to/image.png';
+
+$value = new \eZ\Publish\Core\FieldType\Image\Value(
+    array(
```
Comment (mikadamczyk) [inline docs/api/public_php_api_creating_content.md]:
```suggestion
   [
```

File: docs/api/public_php_api_creating_content.md [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+        'path' => '/path/to/image.png',
+        'fileSize' => filesize( '/path/to/image.png' ),
+        'fileName' => basename( 'image.png' ),
+        'alternativeText' => 'My image'
+    )
```
Comment (mikadamczyk) [inline docs/api/public_php_api_creating_content.md]:
```suggestion
    ]
```

File: docs/api/public_php_api_creating_content.md [hunk 6]
```diff
@@ -0,0 +1,377 @@
...
+}
+
+
+$contentTypeCreateStruct = $contentTypeService->newContentTypeCreateStruct( 'mycontenttype' );
+$contentTypeCreateStruct->mainLanguageCode = 'eng-GB';
+$contentTypeCreateStruct->nameSchema = '<title>';
+$contentTypeCreateStruct->names = array(
+    'eng-GB' => 'My content type'
+);
+$contentTypeCreateStruct->descriptions = array(
```
Comment (mikadamczyk) [inline docs/api/public_php_api_creating_content.md]:
```suggestion
$contentTypeCreateStruct->descriptions = [
```

File: docs/api/public_php_api_creating_content.md [hunk 7]
```diff
@@ -0,0 +1,377 @@
...
+    'eng-GB' => 'Description for my content type',
+);
```
Comment (mikadamczyk) [inline docs/api/public_php_api_creating_content.md]:
```suggestion
];
```

File: docs/api/public_php_api_creating_content.md [hunk 8]
```diff
@@ -0,0 +1,377 @@
...
+```
+
+First, you need to load the `ContentTypeGroup` your `ContentType` will be created in. You do this using [`ContentTypeService::loadContentTypeGroupByIdentifier()`](http://apidoc.ez.no/sami/trunk/NS/html/eZ/Publish/API/Repository/ContentTypeService.html#method_loadContentTypeGroupByIdentifier), which gives you back a `ContentTypeGroup` object. As for content, you then request a `ContentTypeCreateStruct` from the `ContentTypeService`, using `ContentTypeService::newContentTypeCreateStruct()`, with the desired identifier as the argument.
+
+Using the create struct's properties, you can set the Type's properties:
+
+-   the main language (`mainLanguageCode`) for the Type is set to eng-GB,
```
Comment (mikadamczyk) [inline docs/api/public_php_api_creating_content.md]:
```suggestion
-   the main language (`mainLanguageCode`) for the Type is set to `eng-GB`,
```

File: docs/api/public_php_api_managing_content.md [hunk 9]
```diff
@@ -1,421 +1,194 @@
...
+    'eng-GB' => $contentTypeIdentifier,
 );
-$contentTypeCreateStruct->descriptions = array(
-    'eng-GB' => 'Description for my content type',
-);
-```
 
-First, you need to load the `ContentTypeGroup` your `ContentType` will be created in. You do this using [`ContentTypeService::loadContentTypeGroupByIdentifier()`](http://apidoc.ez.no/sami/trunk/NS/html/eZ/Publish/API/Repository/ContentTypeService.html#method_loadContentTypeGroupByIdentifier), which gives you back a `ContentTypeGroup` object. As for content, you then request a `ContentTypeCreateStruct` from the `ContentTypeService`, using `ContentTypeService::newContentTypeCreateStruct()`, with the desired identifier as the argument.
+$titleFieldCreateStruct = $this->contentTypeService->newFieldDefinitionCreateStruct('title', 'ezstring');
+$titleFieldCreateStruct->names = array('eng-GB' => 'Title');
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
$titleFieldCreateStruct->names = ['eng-GB' => 'Title'];
```

File: docs/api/public_php_api_managing_content.md [hunk 10]
```diff
@@ -1,421 +1,194 @@
...
-    return;
-}
-
-
-$contentTypeCreateStruct = $contentTypeService->newContentTypeCreateStruct( 'mycontenttype' );
+$contentTypeCreateStruct = $this->contentTypeService->newContentTypeCreateStruct($contentTypeIdentifier);
 $contentTypeCreateStruct->mainLanguageCode = 'eng-GB';
 $contentTypeCreateStruct->nameSchema = '<title>';
+
 $contentTypeCreateStruct->names = array(
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
$contentTypeCreateStruct->names = [
```

File: docs/api/public_php_api_managing_content.md [hunk 11]
```diff
@@ -1,421 +1,194 @@
...
-    'eng-GB' => 'My content type'
+    'eng-GB' => $contentTypeIdentifier,
 );
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
];
```

File: docs/api/public_php_api_managing_content.md [hunk 12]
```diff
@@ -1,421 +1,194 @@
...
+$contentTypeCreateStruct->addFieldDefinition($titleFieldCreateStruct);
 
-Using the create struct's properties, you can set the Type's properties:
-
--   the main language (`mainLanguageCode`) for the Type is set to eng-GB,
--   the content name generation pattern (`nameSchema`) is set to '&lt;title&gt;': Content items of this type will be named the same as their 'title' field.
--   the human-readable name for your Type is set using the `names` property. You give it a hash, indexed by the locale ('eng-GB') the name is set in. This locale must exist in the system.
--   the same way that you have set the `names` property, you can set human-readable descriptions, again as hashes indexed by locale code.
+try {
+    $contentTypeDraft = $this->contentTypeService->createContentType($contentTypeCreateStruct, array($contentTypeGroup));
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
    $contentTypeDraft = $this->contentTypeService->createContentType(
        $contentTypeCreateStruct, 
        [$contentTypeGroup]
    );
```

File: docs/api/public_php_api_managing_content.md [hunk 13]
```diff
@@ -1,421 +1,194 @@
...
+You can specify more details of the Field definition in the create struct, for example:
 
-```php
-// add a TextLine Field with identifier 'title'
-$titleFieldCreateStruct = $contentTypeService->newFieldDefinitionCreateStruct( 'title', 'ezstring' );
-$titleFieldCreateStruct->names = array( 'eng-GB' => 'Title' );
-$titleFieldCreateStruct->descriptions = array( 'eng-GB' => 'The Title' );
+``` php
+$titleFieldCreateStruct = $this->contentTypeService->newFieldDefinitionCreateStruct('title', 'ezstring');
+$titleFieldCreateStruct->names = array('eng-GB' => 'Title');
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
$titleFieldCreateStruct->names = ['eng-GB' => 'Title'];
```

File: docs/api/public_php_api_managing_content.md [hunk 14]
```diff
@@ -1,421 +1,194 @@
...
+$titleFieldCreateStruct->descriptions = array('eng-GB' => 'The Title');
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
$titleFieldCreateStruct->descriptions = ['eng-GB' => 'The Title'];
```

File: docs/api/public_php_api_managing_repository.md [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+$sectionCreateStruct->identifier = 'newsection';
+$this->sectionService->createSection($sectionCreateStruct);
+```
+
+### Getting Section information
+
+You can use `SectionService` to retrieve Section information such as whether it is in use, or the total number of content assigned to it:
+
+``` php
+$output->writeln(($this->sectionService->isSectionUsed($section) ? "This section is in use." : "This section is not in use."));
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
$output->writeln(($this->sectionService->isSectionUsed($section) ? 'This section is in use.' : 'This section is not in use.'));
```

File: docs/api/public_php_api_managing_repository.md [hunk 16]
```diff
@@ -0,0 +1,214 @@
...
+$output->writeln("Number of Content items in this section: " . $this->sectionService->countAssignedContents($section));
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
$output->writeln('Number of Content items in this section: ' . $this->sectionService->countAssignedContents($section));
```

File: docs/api/public_php_api_managing_repository.md [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+
+[`BookmarkService`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/BookmarkService.php)
+enables you to read, add and remove bookmarks from content.
+
+To view a list of all bookmarks, use [`BookmarkService::loadBookmarks`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/BookmarkService.php#L54):
+
+``` php
+$bookmarkList = $this->bookmarkService->loadBookmarks();
+
+$output->writeln("Total bookmarks: " . $bookmarkList->totalCount);
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
$output->writeln('Total bookmarks: ' . $bookmarkList->totalCount);
```

File: docs/api/public_php_api_managing_repository.md [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
+
+### Getting Language information
+
+To get a list of all Languages in the system use [`LanguageService::loadLanguages`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/LanguageService.php#L81):
+
+``` php
+$languageList = $this->languageService->loadLanguages();
+
+foreach ($languageList as $language) {
+    $output->writeln($language->languageCode . ": " . $language->name);
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
    $output->writeln($language->languageCode . ': ' . $language->name);
```

File: docs/api/public_php_api_creating_content.md [hunk 19]
```diff
@@ -0,0 +1,377 @@
...
+```
+
+But as said before, whatever you provide `setField()` with is sent to the `acceptValue()` method. This method really is the entry point to the input formats a Field Type accepts. In this case, you could have provided setField with either a hash, similar to the one you provided the Image\\Value constructor with, or the path to your image, as a string.
+
+``` php
+$contentCreateStruct->setField( 'image', '/path/to/image.png' );
+
+// or
+
+$contentCreateStruct->setField( 'image', array(
```
Comment (mikadamczyk) [inline docs/api/public_php_api_creating_content.md]:
```suggestion
$contentCreateStruct->setField( 'image',[
```

File: docs/api/public_php_api_managing_repository.md [hunk 20]
```diff
@@ -0,0 +1,214 @@
...
+    ## Workflow
+
+    ### Getting workflow information
+
+    To get information about a specific workflow for a Content item, use `WorkflowServiceInterface::loadWorkflowMetadataForContent`:
+
+    ``` php
+    $workflowMetadata = $this->workflowService->loadWorkflowMetadataForContent($content, $workflowName);
+    foreach ($workflowMetadata->markings as $marking) {
+        $output->writeln($content->getName() . " is in stage " . $marking->name . " in workflow " . $workflowMetadata->workflow->getName());
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
        $output->writeln($content->getName() . ' is in stage ' . $marking->name . ' in workflow ' . $workflowMetadata->workflow->getName());
```

File: docs/api/public_php_api_managing_repository.md [hunk 21]
```diff
@@ -0,0 +1,214 @@
...
+
+### Creating a Language
+
+To create a new language, you need to create a [`LanguageCreateStruct`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/Values/Content/LanguageCreateStruct.php)
+and provide it with the language code and language name.
+Then, use [`LanguageService::createLanguage`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/LanguageService.php#L29) and pass the `LanguageCreateStruct` to it:
+
+``` php
+$languageCreateStruct = $this->languageService->newLanguageCreateStruct();
+$languageCreateStruct->languageCode = "ger-DE";
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
$languageCreateStruct->languageCode = 'ger-DE';
```

File: docs/api/public_php_api_managing_repository.md [hunk 22]
```diff
@@ -0,0 +1,214 @@
...
+$languageCreateStruct->name = "German";
```
Comment (mikadamczyk) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
$languageCreateStruct->name = 'German';
```

File: docs/api/public_php_api_managing_content.md [hunk 23]
```diff
@@ -1,421 +1,197 @@
...
-
-## Assigning Section to content
+    [Copy subtree limit](../guide/config_back_office/#copy-subtree-limit) only applies to operations in the Back Office.
+    It is ignored when copying subtrees using the PHP API.
 
-!!! note "Full code"
+## Trash
 
-    [https://github.com/ezsystems/CookbookBundle/tree/master/Command/AssignContentToSectionCommand.php](https://github.com/docb22/ez-publish-cookbook/tree/master/EzSystems/CookBookBundle/Command/AssignContentToSectionCommand.php)
+To empty the Trash (remove all Locations in Trash), use [`TrashService::emptyTrash`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/TrashService.php#L72),
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
To empty the Trash (remove all Locations in Trash), use [`TrashService::emptyTrash`,](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/TrashService.php#L72)
```

File: docs/api/public_php_api_managing_content.md [hunk 24]
```diff
@@ -1,421 +1,197 @@
...
-Use `ContentService::loadContentInfo()` to get the Content you want to update the Section for.
+## Content Types
 
-**assign section to content**
+### Creating Content Types
 
-``` php
-$section = $sectionService->loadSection( $sectionId );
-```
+To operate on Content Types, you need to make use of [`ContentTypeService`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/ContentTypeService.php).
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
To operate on Content Types, you need to make use of [`ContentTypeService`.](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/ContentTypeService.php)
```

File: docs/api/public_php_api_managing_content.md [hunk 25]
```diff
@@ -1,421 +1,197 @@
...
-Each field's properties are set using the create struct's properties:
-
--   `names` and `descriptions` are set using hashes indexed by the locale code, and with the name or description as an argument.
--   The `fieldGroup` is set to 'content'
--   Fields are ordered using the `position` property, ordered numerically in ascending order. It is set to an integer.
--   The translatable, required and searchable boolean flags are set using their respective property: `isTranslatable`, `isRequired` and `isSearchable`.
+### Copying Content Types
 
-Once the properties for each create struct are set, the field is added to the Content Type create struct using [`ContentTypeCreateStruct::addFieldDefinition()`](http://apidoc.ez.no/sami/trunk/NS/html/eZ/Publish/API/Repository/Values/ContentType/ContentTypeCreateStruct.html#method_addFieldDefinition).
+To copy a Content Type, use [`ContentTypeService::copyContentType`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/ContentTypeService.php#L244):
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
To copy a Content Type, use [`ContentTypeService::copyContentType`:](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/ContentTypeService.php#L244)
```

File: docs/api/public_php_api_managing_content.md [hunk 26]
```diff
@@ -1,421 +1,197 @@
...
-}
+$copy = $this->contentTypeService->copyContentType($contentType);
 ```
 
-The last step is the same as for Content: create a Content Type draft using `ContentTypeService::createContentType()`, with the `ContentTypeCreateStruct` and an array of `ContentTypeGroup` objects are arguments. Then publish the Content Type draft using `ContentTypeService::publishContentTypeDraft()`.
+The copy will automatically be given an identifier based on the original Content Type identifier
+and the copy's ID, for example: `copy_of_folder_21`.
 
-## Deleting Content
+To change the identifier of the copy, use a [`ContentTypeUpdateStruct`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/Values/ContentType/ContentTypeUpdateStruct.php):
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_content.md]:
```suggestion
To change the identifier of the copy, use a [`ContentTypeUpdateStruct`:](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/Values/ContentType/ContentTypeUpdateStruct.php)
```

File: docs/api/public_php_api_managing_repository.md [hunk 27]
```diff
@@ -0,0 +1,214 @@
...
+
+``` php
+$output->writeln(($this->sectionService->isSectionUsed($section) ? 'This section is in use.' : 'This section is not in use.'));
+$output->writeln('Number of Content items in this section: ' . $this->sectionService->countAssignedContents($section));
+```
+
+### Listing content in a Section
+
+To list Content items assigned to a Section you need to make a [query](php_public_api_search.md)
+for Content belonging to this section, by applying the [`SearchService`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/SearchService.php):
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
for Content belonging to this section, by applying the [`SearchService`:](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/SearchService.php)
```

File: docs/api/public_php_api_managing_repository.md [hunk 28]
```diff
@@ -0,0 +1,214 @@
...
+
+## Object states
+
+[Object states](../guide/admin_panel.md#object-states)  enable you to set a custom state to any content.
+States are grouped into Object state groups.
+
+### Creating Object states
+
+To create an Object state group and add Object states to it,
+you need to make use of the [`ObjectStateService`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/ObjectStateService.php):
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
you need to make use of the [`ObjectStateService`:](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/ObjectStateService.php)
```

File: docs/api/public_php_api_managing_repository.md [hunk 29]
```diff
@@ -0,0 +1,214 @@
...
+
+``` php
+$objectStateGroupStruct = $this->objectStateService->newObjectStateGroupCreateStruct('rank');
+$objectStateGroupStruct->defaultLanguageCode = 'eng-GB';
+$objectStateGroupStruct->names = ['eng-GB' => 'rank'];
+$this->objectStateService->createObjectStateGroup($objectStateGroupStruct);
+```
+
+[`ObjectStateService::createObjectStateGroup`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/ObjectStateService.php#L36)
+takes as argument an [`ObjectStateGroupCreateStruct`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/Values/ObjectState/ObjectStateGroupCreateStruct.php),
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
takes as argument an [`ObjectStateGroupCreateStruct`,](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/Values/ObjectState/ObjectStateGroupCreateStruct.php)
```

File: docs/api/public_php_api_managing_repository.md [hunk 30]
```diff
@@ -0,0 +1,214 @@
...
+    ### Applying workflow transitions
+
+    To place a Content item in a workflow, use `WorkflowService::start`:
+
+    ``` php
+    $this->workflowService->start($content, $workflowName);
+    ```
+
+    To apply a transition to a Content item, use `WorkflowService::apply`.
+    Additionally, to you can check if the transition is possible for the given object using `WorkflowService::can`:
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
    Additionally, you can check if the transition is possible for the given object using `WorkflowService::can`:
```

File: docs/api/public_php_api_managing_repository.md [hunk 31]
```diff
@@ -0,0 +1,214 @@
...
+        $this->workflowService->apply($workflowMetadata, $transitionName, 'Please review');
+    }
+    ```
+
+## Bookmarks
+
+[`BookmarkService`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/BookmarkService.php)
+enables you to read, add and remove bookmarks from content.
+
+To view a list of all bookmarks, use [`BookmarkService::loadBookmarks`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/BookmarkService.php#L54):
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
To view a list of all bookmarks, use [`BookmarkService::loadBookmarks`:](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/BookmarkService.php#L54)
```

File: docs/api/public_php_api_managing_repository.md [hunk 32]
```diff
@@ -0,0 +1,214 @@
...
+You can add a bookmark to a Content item by providing its Location object
+to the [`BookmarkService::createBookmark`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/BookmarkService.php#L31) method:
+
+``` php
+$location = $this->locationService->loadLocation($locationId);
+
+$this->bookmarkService->createBookmark($location);
+```
+
+You can remove a bookmark from a Location with [`BookmarkService::deleteBookmark`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/BookmarkService.php#L42):
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
You can remove a bookmark from a Location with [`BookmarkService::deleteBookmark`:](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/BookmarkService.php#L42)
```

File: docs/api/public_php_api_managing_repository.md [hunk 33]
```diff
@@ -0,0 +1,214 @@
...
+
+``` php
+$this->bookmarkService->deleteBookmark($location);
+```
+
+## Languages
+
+### Getting Language information
+
+To get a list of all Languages in the system use [`LanguageService::loadLanguages`](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/LanguageService.php#L81):
```
Comment (MagdalenaZuba) [inline docs/api/public_php_api_managing_repository.md]:
```suggestion
To get a list of all Languages in the system use [`LanguageService::loadLanguages`:](https://github.com/ezsystems/ezpublish-kernel/blob/v7.5.3/eZ/Publish/API/Repository/LanguageService.php#L81)
```

**Comment by DominikaK** [comment]
Comments for the Creating Content page will be handled in https://jira.ez.no/browse/EZP-30774

**Pull request merged**